tests/functional/graphql/test_related.py

```diff
@@ -0,0 +1,203 @@
+            assert {r['corpusItem']['id'] for r in recs} == {"253c3f1e-00c3-4ec2-a03a-3e046ab16652",
+                                                             "3edd22f6-0967-4eda-af23-d06d7cfea722",
+                                                             "b4cbf203-50cd-45fc-97c8-e1fddac2f270"}
+
+    def test_related_end_of_article_article_doesnt_exist(self):
+        with TestClient(app) as client:
+            response = client.post("/", json={
+                    'query': '''
+                        query ($representations: [_Any!]!) {
+                          _entities(representations: $representations) {
+                            ... on SyndicatedArticle {
+                                itemId
+                                publisherUrl
+                                relatedEndOfArticle(count: 3) {
+                                    id
+                                    corpusItem {
+                                      id
+                                    }
+                                }
+                            }
+                          }
+                        }
+                    ''',
+                    'variables': {
+                        'representations': [
+                            {
+                                '__typename': 'SyndicatedArticle',
+                                'itemId': "11111",
+                                'publisherUrl': "xxxxx",
+                            },
+                        ],
+                    },
+                }).json()
+
+            assert not response.get('errors')
+            recs = response['data']['_entities'][0]['relatedEndOfArticle']
+            assert len(recs) == 0
+
+    def test_related_right_rail_article_doesnt_exist(self):
+        with TestClient(app) as client:
+            response = client.post("/", json={
+                'query': '''
+                        query ($representations: [_Any!]!) {
+                          _entities(representations: $representations) {
+                            ... on SyndicatedArticle {
+                                itemId
+                                publisherUrl
+                                relatedRightRail(count: 3) {
+                                    id
+                                    corpusItem {
+                                      id
+                                    }
+                                }
+                            }
+                          }
+                        }
+                    ''',
+                    'variables': {
+                        'representations': [
+                            {
+                                '__typename': 'SyndicatedArticle',
+                                'itemId': "111111",
+                                'publisherUrl': "xxxx",
+                            },
+                        ],
+                    },
+                }).json()
```

Comment:
Suggestion: add a function to generate this json to make tests easier to read by removing duplicate code.

Reply:
Sometimes it's ok to duplicate code to have the whole test case inside the test, it's easier to understand tests that way